src/gui/torrentcontentmodel.cpp

```diff
@@ -113,31 +106,23 @@ namespace
         {
             const QString ext = info.suffix();
             if (!ext.isEmpty()) {
-                const QPixmap pixmap = pixmapForExtension(ext, QSize(32, 32));
-                if (!pixmap.isNull())
-                    return QIcon(pixmap);
+                auto cacheIter = m_iconCache.find(ext);
+                
+                if (cacheIter != m_iconCache.end())
+                    return *cacheIter;
+                
+                QIcon icon = QIcon(pixmapForExtension(ext, QSize(32, 32)));
+                if (!icon.isNull()) {
+                    m_iconCache.insert(ext, icon);
+                    return icon;
```

Comment:
I think we want to avoid returning a null icon. If it's null, it'll exit the if block and hit `return UnifiedFileIconProvider::icon(info);`. So that way it falls back on the behavior of `UnifiedFileIconProvider`.